Index: .
===================================================================
--- .	(revision 86520)
+++ .	(working copy)
@@ -19,7 +19,7 @@
 #endif
 #include <stack>
 #include <limits>
-#include <typeinfo>
+#include <boost/type_index/type_info.hpp>
 #include <boost/shared_ptr.hpp>
 #include <boost/type_traits/is_same.hpp>
 #include <boost/version.hpp>
@@ -144,7 +144,7 @@
     explicit xpression_linker(Traits const &tr)
       : back_stack_()
       , traits_(&tr)
-      , traits_type_(&typeid(Traits))
+      , traits_type_(&boost::type_id<Traits>())
       , has_backrefs_(false)
     {
     }
@@ -310,13 +310,13 @@
     template<typename Traits>
     Traits const &get_traits() const
     {
-        BOOST_ASSERT(*this->traits_type_ == typeid(Traits));
+        BOOST_ASSERT(*this->traits_type_ == boost::type_id<Traits>());
         return *static_cast<Traits const *>(this->traits_);
     }
 
     std::stack<void const *> back_stack_;
     void const *traits_;
-    std::type_info const *traits_type_;
+    boost::type_info const *traits_type_;
     bool has_backrefs_;
 };
 
